nil pointer dereference when parsing multiple config file

setup:

config/strategy.hcl:

```hcl
strategy "target-value" {
  driver = "target-value"
}
```

config/z_nomad.hcl:

```hcl
nomad {
  address = "http://localhost:4646"
  region  = "global"
  token   = "sometoken"
}
```

nomad-autoscaler agent -config config

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16d9b76]

goroutine 1 [running]:
github.com/hashicorp/nomad-autoscaler/agent/config.(*Nomad).merge(0x0, 0xc0003f6e70, 0x27a)
        /Users/huan/workspace/nomad-autoscaler/agent/config/config.go:448 +0x36
github.com/hashicorp/nomad-autoscaler/agent/config.(*Agent).Merge(0xc0000ca1e0, 0xc0004ae6e0, 0xc0004ae6e0)
        /Users/huan/workspace/nomad-autoscaler/agent/config/config.go:376 +0x83b
github.com/hashicorp/nomad-autoscaler/agent/config.loadDir(0x7ffeefbff431, 0x2, 0x2c5cea0, 0xc0000aa9c0, 0x0)
        /Users/huan/workspace/nomad-autoscaler/agent/config/config.go:837 +0x2a6
github.com/hashicorp/nomad-autoscaler/agent/config.Load(0x7ffeefbff431, 0x2, 0xc0000ca140, 0xc0000945d0, 0x7ffeefbff42a)
        /Users/huan/workspace/nomad-autoscaler/agent/config/config.go:797 +0x212
github.com/hashicorp/nomad-autoscaler/agent/config.LoadPaths(0xc000098400, 0x1, 0x1, 0x2, 0x0, 0x0)
        /Users/huan/workspace/nomad-autoscaler/agent/config/config.go:761 +0x105
github.com/hashicorp/nomad-autoscaler/command.(*AgentCommand).readConfig(0xc000094570, 0x70, 0x2741860, 0xc0000b0301, 0xc0000aec40)
        /Users/huan/workspace/nomad-autoscaler/command/agent.go:370 +0x12db
github.com/hashicorp/nomad-autoscaler/command.(*AgentCommand).Run(0xc000094570, 0xc000130060, 0x2, 0x2, 0x0)
        /Users/huan/workspace/nomad-autoscaler/command/agent.go:224 +0x8f
github.com/mitchellh/cli.(*CLI).Run(0xc0000c4000, 0xc000094420, 0x29ca145, 0x7)
        /Users/huan/go/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1cf
main.main()
        /Users/huan/workspace/nomad-autoscaler/main.go:26 +0x29b
```

--- a/agent/config/config.go
+++ b/agent/config/config.go
@@ -354,6 +354,10 @@ func Default() (*Agent, error) {
 
 // Merge is used to merge two agent configurations.
 func (a *Agent) Merge(b *Agent) *Agent {
+	if a == nil {
+		return b
+	}
+
 	result := *a
 
 	if b.EnableDebug {
@@ -432,6 +436,10 @@ func (a *Agent) Validate() error {
 }
 
 func (h *HTTP) merge(b *HTTP) *HTTP {
+	if h == nil {
+		return b
+	}
+
 	result := *h
 
 	if b.BindAddress != "" {
@@ -445,6 +453,10 @@ func (h *HTTP) merge(b *HTTP) *HTTP {
 }
 
 func (n *Nomad) merge(b *Nomad) *Nomad {
+	if n == nil {
+		return b
+	}
+
 	result := *n
 
 	if b.Address != "" {
@@ -485,6 +497,10 @@ func (n *Nomad) merge(b *Nomad) *Nomad {
 }
 
 func (t *Telemetry) merge(b *Telemetry) *Telemetry {
+	if t == nil {
+		return b
+	}
+
 	result := *t
 
 	if b.StatsiteAddr != "" {
@@ -555,6 +571,10 @@ func (t *Telemetry) merge(b *Telemetry) *Telemetry {
 }
 
 func (p *Plugin) merge(o *Plugin) *Plugin {
+	if p == nil {
+		return o
+	}
+
 	m := *p
 
 	if len(o.Name) != 0 {
@@ -571,6 +591,10 @@ func (p *Plugin) merge(o *Plugin) *Plugin {
 }
 
 func (p *Plugin) copy() *Plugin {
+	if p == nil {
+		return nil
+	}
+
 	c := *p
 	if i, err := copystructure.Copy(p.Config); err != nil {
 		panic(err.Error())
@@ -581,6 +605,10 @@ func (p *Plugin) copy() *Plugin {
 }
 
 func (p *Policy) merge(b *Policy) *Policy {
+	if p == nil {
+		return b
+	}
+
 	result := *p
 
 	if b.Dir != "" {
@@ -596,6 +624,10 @@ func (p *Policy) merge(b *Policy) *Policy {
 }
 
 func (pw *PolicyEval) merge(in *PolicyEval) *PolicyEval {
+	if pw == nil {
+		return in
+	}
+
 	result := *pw
 
 	if in.AckTimeout != 0 {

--- a/agent/config/config_test.go
+++ b/agent/config/config_test.go
@@ -246,6 +246,16 @@ func TestAgent_Merge(t *testing.T) {
 	assert.ElementsMatch(t, expectedResult.APMs, actualResult.APMs)
 	assert.ElementsMatch(t, expectedResult.Targets, actualResult.Targets)
 	assert.ElementsMatch(t, expectedResult.Strategies, actualResult.Strategies)
+
+	// Test merge on nil config.
+	var nilCfg *Agent
+	actualResult = nilCfg.Merge(baseCfg)
+	assert.Equal(t, baseCfg, actualResult)
+
+	// Test merge on empty config.
+	emptyCfg := &Agent{}
+	actualResult = emptyCfg.Merge(baseCfg)
+	assert.Equal(t, baseCfg, actualResult)
 }
 
 func TestAgent_parseFile(t *testing.T) {

